app/uk/gov/hmrc/gform/dms/DmsSubmissionWithAttachmentsRequestInterpreter.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.dms
+
+import java.nio.file.Files.readAllBytes
+import java.nio.file.Paths
+
+import play.api.libs.Files.TemporaryFile
+import play.api.libs.json.{ JsError, JsSuccess, Json }
+import play.api.mvc.{ MultipartFormData, Request }
+
+object DmsSubmissionWithAttachmentsRequestInterpreter {
+  def apply(
+    request: Request[MultipartFormData[TemporaryFile]]): Either[String, (String, List[Array[Byte]], DmsMetadata)] = {
+
+    val maybeHtml: Option[Seq[String]] = request.body.dataParts.get("html")
+    val maybeMetadata: Option[DmsMetadata] = Json
+      .toJson(request.body.dataParts.filterKeys(_ == "html").mapValues(_.mkString("")))
```

Comment:
this is a bit messy - I am sending the HTML content and the DMS meta data in the dataparts of the MultipartFormData request.
So I filter out the html to just get the DMS meta keys
if you know a better way to do this, let me know